### DISCUSSION TITLE
MarpWebEditor is Here! A New Web-Based Editor for Marp Enthusiasts!

### DISCUSSION THREAD

**Discussion by link2004** [marp-team/marp#531] [Show and tell]
Hey Marp enthusiasts!
I'm excited to announce the launch of MarpWebEditor.app, a new tool I've developed to enhance your Marp experience.
This web-based editor comes packed with some cool features:

You can edit your Markdown and CSS in real-time, seeing your changes instantly
Your work is safe with built-in saving for both MD and CSS files
When you're ready to share, you can export your presentations as HTML, PDF, or even PowerPoint (PPTX) files
And don't worry about screen size - the app is fully responsive!

I've put a lot of work into making this tool as user-friendly and useful as possible. I'd love for you all to check it out and let me know what you think. Your feedback would be invaluable in helping me improve it further.
Feel free to give MarpWebEditor.app a spin and share your thoughts. Looking forward to hearing from you!

Click Here👇
[MarpWebEditor](https://marpwebeditor.app/)

**Comment by greyovo** [comment]
Great work! Some suggestions:
1. The font of the editor could be monospaced fonts  (e,g, Consolas)
2. Is it possible to implment a two-column layout?

**Reply by link2004** [reply]
Thank you for your suggestions. Let me address your points:

1. Regarding the editor font: The font is already monospaced. Could you provide more details about your environment or what specific issue you're experiencing with the font?
![image](https://github.com/user-attachments/assets/e4da63f1-52ae-4f33-84bc-8047ebb91599)

2. About the two-column layout: Are you referring to a two-column layout for slide previews? While it's technically possible, we haven't considered implementing this feature at the moment. However, you can adjust the layout by moving the central boundary line left or right to resize the different sections of the interface.

Is there anything specific you'd like to improve about the current layout or functionality? I'd be happy to discuss further options or alternatives if you have particular needs in

**Reply by greyovo** [reply]
@link2004 Thank you for your detailed reply!
1. About the font: Yes, the font I saw is consistent with your picture. However, the readability of this font is just not ideal. I hope that modern fonts such as Consolas, Meno, or Monaco can be used, which should be better.
2. Two-column list: I hope to expand the existing syntax of Marp to display content in two columns, like the following:
> NOTE: This may not be the final syntax, just for showcase :)

```md
## My Title

<!-- @left -->
1. Item 1
2. Item 2
4. Item 3

<!-- @right-->
1. Item 4
2. Item 5
3. Item 6
```
Then render as:

![Clip_2024-09-10_20-10-47](https://github.com/user-attachments/assets/61dc6445-80ed-4c1a-93d9-a01743a32420)

The double-column list is more like a functional extension, and there has been some discussion on how to implement it. I am very interested in your project. Is it open source? If I have time I would like to contribute

**Reply by link2004** [reply]
Thanks for the reply!
MarpWebEditor is just an Editor, so it relies on marp-core's functionality for slide conversion.
Internally, we use @marp-team/marp-core to convert Markdown to html and draw it.
If @marp-team/marp-core adds a two-column list display function, MarpWebEditor will be able to support it!

Thanks for your feedback about the fonts!
We are considering adding a feature to change the font!

(I use the Deepl translation)

**Comment by ChiMaoShuPhy** [comment]
Very helpful and great work!  I am wondering is it possible to deploy this web-editor locally?

**Comment by yhatt** [comment]
Notice by Marp team:

- This service displays a copyright notice "&copy; Marp team" which is misleading. Marp team has no involvement in its development.
- A different service by the same developer has been reported to contain vulnerabilities that could allow the collection of personal information of service users. At this time, it has not been confirmed whether such vulnerabilities exist in that service, but please use it at your own risk.
